### PR TITLE
main: Add informative message what do to if RecurrsionError occurs.

### PR DESCRIPTION
**Pull request by htgoebel**
See #4406

### CODE REVIEW

**Review by bjones1 — APPROVED** [review]

**Review by Legorooj — APPROVED** [review]
Nice work! EDIT: see below

**Comment by Legorooj** [comment]
However, the new test fails - can you fix this? Please also fix the lint error as well.

**Comment by htgoebel** [comment]
Interesting to see the recursion on Python 3.6 is deeper than on he other ones.

**Comment by Legorooj** [comment]
@htgoebel can you rebase off develop? That should fix the test failures.

**Comment by htgoebel** [comment]
@Legorooj Will do, but I'm curious: Current develop head only adds "hooks: Stop hooks collecting bytecode files". What is the reason why you expect this to stop the worker-crashes?

**Comment by Legorooj** [comment]
Current development head also reverts your accidental commit AFAIK - it worked for other PRs with the same issue (I think it's the same issue anyway)

**Review by Legorooj — CHANGES_REQUESTED** [review]
Just one _little_ thing: can you please fix the following filename:
```diff
+ _recursion_too_deep_message.py
- _recursion_to_deep_message.py
```

**Comment by htgoebel** [comment]
The tests for this case are inconsistently failing on Win-64 Py 3.7 and 3.8 only: The respective worker is crashing. But this does not happen if only running these tests, see https://ci.appveyor.com/project/htgoebel/pyinstaller/builds/35176564. Since I have not time for debugging CI platform annoyances, I'll just remove the tests. @Legorooj Maybe you want to keep them as a base for tests when moving to modulegraph2.

**Comment by Legorooj** [comment]
@htgoebel I did some digging and it looks like this is an issue with pytest-xdist's memory handling. I've applied the same fix that SciPy did so we'll merge if that works, and if not, keep digging.

**Comment by htgoebel** [comment]
Since the pinning pytest-xdist did not help, I'm going to remove these test.

**Comment by htgoebel** [comment]
Or better: I'm xfailing them for Wondoze.

**Review by Legorooj — APPROVED** [review]

File: tests/unit/test_recursion_limit.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        '--workpath', str(tmpdir.join("build")),
+        '--path', str(tmpdir),
+    ]
+
+    pyi_args = [script] + default_args
+    PYI_CONFIG = configure.get_config(upx_dir=None)
+    PYI_CONFIG['cachedir'] = str(tmpdir)
+
+    with pytest.raises(SystemExit) as execinfo:
+        pyi_main.run(pyi_args, PYI_CONFIG)
```
Comment (Legorooj) [inline tests/unit/test_recursion_limit.py:83]: This clause isn't catching the RecursionError for some reason...

**Review by Legorooj — CHANGES_REQUESTED** [review]

**Pull request merged**